### DISCUSSION TITLE
Why can't I access the local file system in browser mode?

### DISCUSSION THREAD

**Discussion by xuxiaokang0723** [AvaloniaUI/Avalonia#17799] [Q&A]
### Describe the bug

![image](https://github.com/user-attachments/assets/acebde70-8168-47e0-8bb6-c7dfe0043478)
In the following code, I want to access a separate configuration file in the local file system. When I am in desktop mode, there is no problem. Now I switch to browser mode and it cannot be accessed.

### To Reproduce

!

### Avalonia version

11.2.2

### OS

WebAssembly

**Comment by Gillibald** [comment]
The browser sandbox has no direct access to the file system

**Reply by thevortexcloud** [reply]
To add to this: Giving the browser direct file system access would be a terrible idea. It would allow every website to steal data off your device as soon as you loaded the page. Which is why no modern browser allows for this.

You can instead store your file on the server as a static file that is loaded by your app. Or you can prompt the user to upload the file. You can also generate the file at runtime and store it in one of the many browser storage locations if it needs to be user configurable. EG

https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API
https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API

Avalonia also has some APIs for dealing with file pickers.

https://docs.avaloniaui.net/docs/concepts/services/storage-provider/

tl;dr Web browsers have much stricter security requirements than desktop apps because they are constantly loading new untrusted code from the internet.

EDIT: Another option I forgot to mention: Don't use a file at all. Instead make a web API (EG using ASP Core) and query that instead.
